src/models/edge.ts

```diff
@@ -293,10 +338,105 @@ abstract class Edge<N extends INodeBase, E extends IEdgeBase> implements IEdge<N
         return null;
     }
   }
+
+  setData(data: E): void;
+
+  setData(callback: (edge: IEdge<N, E>) => E): void;
+
+  setData(arg: E | ((edge: IEdge<N, E>) => E)) {
```

Comment:
```suggestion
  setData(data: E): void;
  setData(callback: (edge: IEdge<N, E>) => E): void;
  setData(arg: E | ((edge: IEdge<N, E>) => E)) {
```

Maybe it makes sense to group them like this to have a better view of function overloads.